app/controllers/ActivityLogController.java

```diff
@@ -0,0 +1,45 @@
+package controllers;
+
+import javax.inject.Inject;
+
+import play.db.jpa.Transactional;
+import play.mvc.Result;
+import services.ActivityLogService;
+
+/**
+ * The type ActivityLog controller.
+ */
+
+public class ActivityLogController extends BaseController {
+
+	private ActivityLogService service;
+	
+	/**
+	 * Sets service.
+	 *
+	 * @param service the service
+	 */
+	@Inject
+	public void setService(ActivityLogService service){
+		this.service = service;
+	}
+	
+	/**
+	 * Gets all activity logs.
+	 * 
+	 * @return the all activity logs
+	 */
+	@Transactional(readOnly = true)
+	public Result getAllActivityLogs() {
+		return wrapForAdmin(() -> this.service.getAllActivityLogs());
+	}
+	
+	/**
+	 * Save activity log.
+	 * 
+	 * @return the result
+	 */
+	public Result postActivityLog(String log){	
+		return wrapForUser(() -> this.service.postActivityLog(log));

```

Comment:
You can get the current user here using `User user = this.cache.get(session("uid"))`, and pass it to the service method.

Reply:
Maybe this should not be an API at all. All the calls can be made from all other services to the Activity Log Service. Inject the CacheApi to gain access to current user.